handbook/engineering/languages/exposing-services.md

```diff
@@ -0,0 +1,30 @@
+
+### Exposing services
+
+In Go, that looks like this
+```
+http.ListenAndServe(":80", nil)
+```
+The above code will bind to all TCP interfaces. Since Kubernetes does support dual-stack IPv4 & IPv6 our services should bind to all interfaces (we do not currently have IPv6 services).
+
+If you must specify an ip address to expose your service on, choosing `0.0.0.0:port` is typically a good choice. What this does can be OS and platform-specific.
+
+Binding to `localhost:port` or `127.0.0.1:port` is binding to a local-only interface that constrained to the same "host". In Kubernetes, other containers within the Pod may still communicate with this service AND you may port-forward this container and associated service
+[(Why?)](#How-can-I-port-forward-a-local-only-service?).
+
+This may be preferred in a sidecar pattern where you do not want a container accessible outside a Pod or when you don't want expose your laptop to the cofeeshop wifi but generally, code should not be merged that binds to `localhost` or `127.0.0.1`.
```

Comment:
Might want to point out that we have a CI check for this: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/dev/check/no-localhost-guard.sh?subtree=true

Reply:
Hmm... why did this not fail this PR then?...
https://github.com/sourcegraph/sourcegraph/pull/12758/files#r466173773
.................. 🙈

Reply:
Yeah not too sure about the context of that check in particular, but I've ran into it failing my build before and it's a related topic so 🤷